src/server/document-controller.js

```diff
@@ -60,7 +62,7 @@ class DocumentController {
         throw new NotFound();
       }
 
-      return res.redirect(301, urls.getDocUrl(doc.key, doc.slug));
+      return res.redirect(301, urls.getDocUrl(doc.key, doc.slug || UNDEFINED_SLUG));
```

Comment:
Rather than using this magic string you could do here:

~~~
if (!doc.slug) {
  const mappedDoc = await this.clientDataMapper.mapDocOrRevision(doc, req.user);
  return this.pageRenderer.sendPage(req, res, PAGE_NAME.doc, { currentDocOrRevision: mappedDoc, type: DOCUMENT_TYPE.document });
}

Reply:
Oh I second that!

Reply:
Done. It's a bit inconsistent though.